libs/file_processing/file_processing_core.py

```diff
@@ -150,11 +151,17 @@ def do_process_user_file_chunks(
 
     # there are several failure modes and success modes, information for what to do with different
     # files percolates back to here.  Delete various database objects accordingly.
-    more_ftps_to_remove, number_bad_files = upload_binified_data(
+    more_ftps_to_remove, number_bad_files, earliest_time_bin, latest_time_bin = upload_binified_data(
         all_binified_data, error_handler, survey_id_dict
     )
     ftps_to_remove.update(more_ftps_to_remove)
 
+    # Update the data quantity stats, if it actually processed any files
+    if len(files_to_process) > 0:
+        calculate_data_quantity_stats(participant,
+                                      earliest_time_bin_number=earliest_time_bin,
+                                      latest_time_bin_number=latest_time_bin)
+
```

Comment:
:warning::question: Not sure if we'll ever hit this case, but could be worth protecting it anyway: what if `earliest_time_bin` or `latest_time_bin` is `None` here? This could cause potentially massive slowdowns in processing data. Not sure the best way to handle, but could be a Sentry error and proceed without calculating data quantity stats?